### PR TITLE
feat(config): add config support

### PR DESCRIPTION
**Pull request by IlyaSurmay**
Added config support.
Test config - https://gist.github.com/IlyaSurmay/84e507d7e9def101ca1c45343982c329 (was tested with ngx-bootstrap)

This should not break existing functionality because CLI options like `-p`, `--watch` have bigger priority than config ones.

todo: set valid `es2015` field in `package.json` if module has multiple entry points

### CODE REVIEW

File: src/bin/ngm-cli.ts [hunk 1]
```diff
@@ -104,10 +105,10 @@ if (cli.input.length === 0) {
 }
 
 // project flag is mandatory for now
-if (!cli.flags.project) {
-  console.error('Please provide path to your projects source folder, `-p DIR` ');
-  process.exit(1);
-}
+// if (!cli.flags.project) {
```
Comment (valorkin) [inline src/bin/ngm-cli.ts:108]: You will have to add another check

File: src/commands/build.command.ts [hunk 2]
```diff
@@ -156,8 +156,32 @@ export function buildCommand({project, verbose, clean, local, main, watch, skipB
 }
 
 export function buildTsRun(cli) {
-  const {project, watch, verbose, clean, local, skipBundles} = cli.flags;
-  let main = cli.flags.main || 'index.ts';
+  const config = cli.flags.config ? require(path.resolve(cli.flags.config)) : {};
```
Comment (valorkin) [inline src/commands/build.command.ts]: Dont inline require

File: src/commands/build.command.ts [hunk 3]
```diff
@@ -156,8 +156,32 @@ export function buildCommand({project, verbose, clean, local, main, watch, skipB
...
+  let {src, entryPoints, watch, verbose, clean, local, skipBundles} = config;
+  const main = cli.flags.main || 'index.ts';
```
Comment (valorkin) [inline src/commands/build.command.ts]: Main file from config?

File: src/commands/build.command.ts [hunk 4]
```diff
@@ -156,8 +156,32 @@ export function buildCommand({project, verbose, clean, local, main, watch, skipB
...
+  const project = cli.flags.project || src;
+  verbose = cli.flags.verbose || verbose;
+  watch = cli.flags.watch || watch;
+  clean = cli.flags.clean || clean;
+  skipBundles = cli.flags.skipBundles || skipBundles;
+
+  if (!project && (!entryPoints || entryPoints.length)) {
```
Comment (valorkin) [inline src/commands/build.command.ts]: Entry point is a relative path to file, not folder

File: src/commands/build.command.ts [hunk 5]
```diff
@@ -156,8 +156,32 @@ export function buildCommand({project, verbose, clean, local, main, watch, skipB
...
+    console.error('Please provide path to your projects source folder, `-p DIR` or specify `src` or `entryPoints` in config');
+    process.exit(1);
+  }
+
+  if (entryPoints && entryPoints.length) {
+    const commands = [];
+    entryPoints.forEach((entryPoint: string) => {
```
Comment (valorkin) [inline src/commands/build.command.ts]: Use .map

File: src/commands/build.command.ts (hunk 4 above)
Comment (valorkin) [reply src/commands/build.command.ts]: So there are should be an array of folders and each folder should have at least one entry point, like ng cli has

File: src/commands/build.command.ts [hunk 6]
```diff
@@ -156,8 +156,32 @@ export function buildCommand({project, verbose, clean, local, main, watch, skipB
...
+      commands.push({
+        title: `Build ${entryPoint}`,
+        task: () => buildCommand({project: entryPoint, verbose, clean, local, main, watch, skipBundles})
+      })
+    });
+    const command = Promise.resolve().then(() => new Listr(commands, {renderer: verbose ? 'verbose' : 'default'}));
```
Comment (valorkin) [inline src/commands/build.command.ts]: Do you really need promise.resolve?

File: src/utils/tasks-watch.ts [hunk 7]
```diff
@@ -21,6 +24,9 @@ export function tasksWatch({project, tasks, watch}){
     }
 
     isRunning = true;
+    tasks.tasks.forEach((task: any, i: number) => {
```
Comment (valorkin) [inline src/utils/tasks-watch.ts]: Tasks.tasks :(

**Review by valorkin — CHANGES_REQUESTED** [review]

File: src/bin/ngm-cli.ts (hunk 1 above)
Comment (IlyaSurmay) [reply src/bin/ngm-cli.ts:108]: It's there https://github.com/valor-software/npm-submodules/pull/17/files#diff-c3ca6689a5afcaf7010b439a1d03bdc3R233

File: src/commands/build.command.ts (hunk 2 above)
Comment (IlyaSurmay) [reply src/commands/build.command.ts]: fs.readFile ?

**Pull request merged**